zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -289,12 +289,34 @@ impl<P: consensus::Parameters> RusqliteMigration for WalletMigration2<P> {
                     }
                 }
 
-                add_account_internal::<P, WalletMigrationError>(
-                    &self.params,
-                    transaction,
-                    "accounts_new",
-                    account,
-                    &ufvk,
+                let ufvk_str: String = ufvk.encode(&self.params);
+                let address_str: String = ufvk.default_address().0.encode(&self.params);
+
+                // This migration, and the wallet behaviour before it, stored the default
+                // transparent address in the `accounts` table. This does not necessarily
+                // match the transparent receiver in the default Unified Address. Starting
+                // from `AddressesTableMigration` below, we no longer store transparent
+                // addresses directly, but instead extract them from the Unified Address
+                // (or from the UFVK if the UA was derived without a transparent receiver,
+                // which is not the case for UAs generated by this crate).
```

Comment:
I don't understand this comment fully; it corresponds to the code below but I don't think this is doing the right thing. We still need to retain the transparent address that was previously generated, so we shouldn't be overwriting it with the taddr that's the default for the UFVK.

Reply:
Per our earlier discussion, what I will do is ensure that the `transparent_address` column is indeed storing the address we expect it to be, and then continue to delete the column. We will instead have a new API that provides a list of all the transparent receivers that should be scanned for funds; it will be composed of all transparent receivers for all UAs in the `addresses` table, plus the default transparent address under each account (because the autoshielding code was generating that transparent address, and we cannot predict whether it was generated for accounts other than account 0 that the SDKs were using, so we have to scan it in perpetuity).